pandas/tests/reshape/test_melt.py

```diff
@@ -212,6 +212,33 @@ def test_multiindex(self):
         res = self.df1.melt()
         assert res.columns.tolist() == ['CAP', 'low', 'value']
 
+    @pytest.mark.parametrize("col", [
+        pd.Series(pd.date_range('2010', periods=5, tz='US/Pacific')),
+        pd.Series(["a", "b", "c", "a", "d"], dtype="category")])
+    @pytest.mark.parametrize("pandas_dtype_value", [True, False])
+    def test_pandas_dtypes_id_var(self, col, pandas_dtype_value):
+        # GH 15785
+        # Pandas dtype in the id
+        df = DataFrame({'klass': range(5),
+                        'col': col,
+                        'attr1': [1, 0, 0, 0, 0]})
+        if pandas_dtype_value:
```

Comment:
can you move what this does into the parameterize, maybe by using a fixture. this defeats the purpose of being able to look at the parameterize and see what cases are being tested